### PR TITLE
feat : migrate staking to new design

### PR DESCRIPTION
**Pull request by Teja2045**
#442

### CODE REVIEW

File: frontend/src/pages/StakingPage.jsx [hunk 1]
```diff
@@ -1,24 +1,51 @@
...
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Paper, Typography } from '@mui/material';
+import Validators from './Validators';
+import { resetDefaultState as stakingReserDefaultState} from '../features/staking/stakeSlice';
+import { resetDefaultState as  distributionResetDefaultState} from '../features/distribution/distributionSlice';
+
 
 export default function MultisigPage() {
```
Comment (aleem1314) [inline frontend/src/pages/StakingPage.jsx:12]: MultisigPage?

File: frontend/src/pages/StakingPage.jsx [hunk 2]
```diff
@@ -1,24 +1,51 @@
...
+                    <Typography color="text.primary" variant="h6" fontWeight={600}>
+                        STAKING
+                    </Typography>
+                    <SelectNetwork 
+                        onSelect={(name) => {handleOnSelect(name)}}
+                        networks={Object.keys(nameToChainIDs)}
+                        defaultNetwork={currentNetwork?.length > 0 ? currentNetwork.toLowerCase().replace(/ /g, "") : "cosmoshub"}
+                    />
+                </Box>
+                <div sx={{justifyContent: "center",display: "flex",mr: 1,}}>
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/pages/StakingPage.jsx]: done

File: frontend/src/pages/StakingPage.jsx (hunk 1 above)
Comment (Teja2045) [reply frontend/src/pages/StakingPage.jsx]: done

File: frontend/src/pages/Validators.js [hunk 3]
```diff
@@ -113,12 +126,15 @@ export default function Validators() {
         break;
       case "redelegate":
         let isValidRedelegation = false;
-        if (delegations?.delegations.length > 0) {
-          for (let i = 0; i < delegations?.delegations.length; i++) {
-            let item = delegations?.delegations[i];
+        if (delegations?.delegations.delegations.length > 0) {
```
Comment (charymalloju) [inline frontend/src/pages/Validators.js:129]: you can assign to delegations to one variable and you can take it from variable

File: frontend/src/components/ActiveValidators.jsx [hunk 4]
```diff
@@ -14,14 +14,15 @@ import { useSelector } from "react-redux";
 import { useTheme } from "@emotion/react";
 
 export function ActiveValidators(props) {
-  const { onMenuAction } = props;
-
-  const validators = useSelector((state) => state.staking.validators);
+  const { onMenuAction, chainID } = props;
+  const validators = useSelector(
+    (state) => state.staking.chains[chainID].validators
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/ActiveValidators.jsx:19]: the chain object is initialized with 
resetDefaultState(chainIDs) reducer.. 
the reducer is called whenever there is a change in wallet state

File: frontend/src/components/Delegations.jsx [hunk 5]
```diff
@@ -161,11 +163,17 @@ export function MyDelegations(props) {
                     %
                   </StyledTableCell>
                   <StyledTableCell align="center">
-                    {parseFloat(row.delegation.shares) /
-                      10 ** currency?.coinDecimals}
+                    {(
+                      parseFloat(row.delegation.shares) /
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/Delegations.jsx:167]: done

File: frontend/src/components/Delegations.jsx [hunk 6]
```diff
@@ -161,11 +163,17 @@ export function MyDelegations(props) {
...
+                      10 ** currency?.coinDecimals
+                    )
+                      .toFixed(3)
+                      .toLocaleString()}
                   </StyledTableCell>
                   <StyledTableCell align="center">
-                    {rewardsP[row.delegation.validator_address]?.toFixed(6)}
+                    {rewardsP[row.delegation.validator_address]
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/Delegations.jsx:174]: done

File: frontend/src/components/DialogRedelegate.jsx [hunk 7]
```diff
@@ -1,206 +1,235 @@
...
-    for (const v in active) {
-        // if (v !== validator?.operator_address)
-            result.push({
-                addr: v,
-                label: active[v].description.moniker
-            })
-    }
+  for (const v in active) {
+    // if (v !== validator?.operator_address)
+    result.push({
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/DialogRedelegate.jsx]: done

File: frontend/src/components/DialogRedelegate.jsx [hunk 8]
```diff
@@ -1,206 +1,235 @@
...
+        addr: v,
+        label: inactive[v].description.moniker,
+      });
+  }
 
-    return result
+  return result;
 }
 
 function parseDelegation(delegations, validator, currency) {
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/DialogRedelegate.jsx]: done

File: frontend/src/components/DialogRedelegate.jsx [hunk 9]
```diff
@@ -1,206 +1,235 @@
...
+    validator,
+    delegations,
+    currency,
+    onRedelegate,
+    loading,
+    authzLoading,
+  } = props;
+
+  const targetValidators = parseValidators(active, inactive, validator);
+
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/components/DialogRedelegate.jsx]: done

File: frontend/src/features/staking/stakeSlice.js [hunk 10]
```diff
@@ -470,19 +482,24 @@ export const stakeSlice = createSlice({
         state.chains[chainID].validators.errMsg = "";
 
         const activeSort = Object.fromEntries(
-          Object.entries(state.chains[chainID].validators.active).sort(([, a], [, b]) => {
-            return b.tokens - a.tokens;
-          })
+          Object.entries(state.chains[chainID].validators.active).sort(
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/features/staking/stakeSlice.js:489]: done

File: frontend/src/pages/StakingPage.jsx [hunk 11]
```diff
@@ -1,24 +1,63 @@
...
-                sx={{
-                    justifyContent: "end",
-                    display: "flex",
-                    mr: 1,
-                }}
-            >
-            </Box>
+  useEffect(() => {
+    dispatch(stakingResetDefaultState(Object.keys(networks)));
+    dispatch(distributionResetDefaultState(Object.keys(networks)));
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/pages/StakingPage.jsx]: done

File: frontend/src/pages/Validators.js (hunk 3 above)
Comment (Teja2045) [reply frontend/src/pages/Validators.js]: done

File: frontend/src/pages/StakingPage.jsx [hunk 12]
```diff
@@ -1,24 +1,63 @@
...
+          </Typography>
+          <SelectNetwork
+            onSelect={(name) => {
+              handleOnSelect(name);
+            }}
+            networks={Object.keys(nameToChainIDs)}
+            defaultNetwork={
+              currentNetwork?.length > 0
+                ? currentNetwork.toLowerCase().replace(/ /g, "")
+                : "cosmoshub"
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/pages/StakingPage.jsx]: done

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**